### PR TITLE
Create java-guide.md

### PR DESCRIPTION
**Pull request by NickGrantX**
Create list of useful Java/Maven/Spring/Hibernate resources - how-to guides, tutorials and explainer guides

### CODE REVIEW

**Comment by NickGrantX** [comment]
[Create list of useful Java/Maven/Spring/Hibernate resources - how-to guides, tutorials and explainer guides.](https://trello.com/c/RsmjIPWD/701-create-list-of-useful-java-maven-spring-hibernate-resources-how-to-guides-tutorials-and-explainer-guides)

**Review by AdamHawtin — APPROVED** [review]
Fun for the whole family

File: java-guide.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+Then, we have used an open-source plugin from Spotify called `dockerfile-maven-plugin` which we use to build a Docker image from the Dockerfile and the JAR file which will have been created in the `target` directory. It's this Docker image which is uploaded to the Google Container Registry (GCR) and will be used by Kubernates to execute our application on the Google Cloud Platform. This plugin only creates the image in your _local_ respository - it doesn't push it up to the GCR.
+
+If you want to do further reading about Maven, the [homepage for the project is located here](http://maven.apache.org/)
+
+I would advise against using the raw Maven reference documentation, because most of the 'clever stuff' we do with Maven comes from the Spring Boot Starter Parent project and the plugins we've used. While the documentation here might not seem very exhaustive, it does constitute most of what you need to know Maven is able to do, and to delve deeper into Maven's vast complexity would probably only serve to confuse and intimidate.
+
+### Spring Boot
+Spring's 'quick start` guide might be worth having a play with. [You can find it here](https://spring.io/quickstart)
```
Comment (AdamHawtin) [inline java-guide.md]: Mismatched ' with a ` here

**Comment by LeoKHoward** [comment]
I would say that this is a good foundation especially for someone such as myself who just accepts that Maven/Spring/Mockito etc. are involved without actually taking the time to understand why they are there. It would be really useful if all of us could start contributing to this guide with useful additions. Hopefully we can even incorporate things like a glossary going forwards. Thanks for making this though NG!

File: java-guide.md (hunk 1 above)
Comment (NickGrantX) [reply java-guide.md]: Fixed

**Review by gavinleeedwards — APPROVED** [review]
A lot of good stuff there and a really good starting point 👍

**Pull request merged**